ebu_tt_live/node/consumer.py

```diff
@@ -91,6 +91,20 @@ def __init__(self, node_id, reference_clock, segment_length, discard, sequence_i
         self._segment_counter = 1
         self._sequence_identifier = sequence_identifier
         self._discard = discard
+        
+        if init_document is not None:
+            # Create sequence from init document, in order to immediately start document output
+            log.info('Creating document sequence from init document {}'.format(
+                init_document
+            ))
+            with open(init_document, 'r') as xml_file:
+                xml_content = xml_file.read()
+            xml_doc = EBUTT3Document.create_from_xml(xml_content)
+            
+            self._sequence = EBUTT3DocumentSequence.create_from_document(xml_doc, verbose=self._verbose)
```

Comment:
At this stage, we are duplicating code that is run by `process_document()` - did you consider simply calling `self.process_document(xml_doc)` instead?

Reply:
Actually I considered this, but the call to `EBUTT3DocumentSequence.add_document` then had the undesired consequence that the init document's sequence number is "burnt" i.e. cannot be used a second time by any real input document. Also the log messages would indicate that a real document had been received, instead of refering to the init document. So I duplicated/adjusted the relevant code part.

But in 564c6021d4f98a67188edbd1696fa7e66d56f81c I just moved the common parts into a new function to have a SPOF here.

Reply:
Ah, that's really interesting, I hadn't spotted there's a code path for setting up the sequence without setting any expectation in code about sequence number. This looks good, thank you.